### PR TITLE
Merge contexts builders' responsibilities into contexts

### PR DESCRIPTION
**Pull request by chives**

### CODE REVIEW

**Comment by chives** [comment]
@fsi-open

**Comment by JarekW** [comment]
:+1:

**Comment by chives** [comment]
@fsi-open

File: Admin/CRUD/Context/BatchElementContext.php [hunk 1]
```diff
@@ -41,68 +37,63 @@ class BatchElementContext implements ContextInterface
     protected $indexes;
 
     /**
-     * @param array $requestHandlers
+     * @param HandlerInterface[]|array $requestHandlers
```
Comment (szymach) [inline Admin/CRUD/Context/BatchElementContext.php]: So this suggests that the $requestHandlers variable can be either an object implementing HandlerInterface or an array, but typehinting clearly requires it to be an array. Am I missing something?

**Comment by szymach** [comment]
:+1:

**Comment by rn0** [comment]
:+1:

**Pull request merged**